jbehave-core/src/main/java/org/jbehave/core/steps/Steps.java

```diff
@@ -207,7 +207,9 @@ public Object instance() {
     }
 
     private boolean isAnnotationPresent(Class<? extends Annotation> type, Collection<StepCandidate> candidates) {
-        return candidates.stream().map(StepCandidate::getMethod).allMatch(m -> m.isAnnotationPresent(type));
+        return candidates.stream()
+                .map(StepCandidate::getMethod)
+                .allMatch(m -> m.isAnnotationPresent(type) || m.getDeclaringClass().isAnnotationPresent(type));
```

Comment:
Does it makes sense combine this walk along with the methods and the one https://github.com/valfirst/jbehave-core/pull/82/files#diff-3ad20dacc88624d47da147f0012009dcR902-R903
And already have not a list of methods but Map<Condition, Method>

Reply:
that would require to change overall jbehave architecture: everything is loaded dynamically